polaris-core/src/main/java/org/apache/polaris/core/policy/validator/GenericPolicyValidator.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.policy.validator;
+
+import com.google.common.base.Preconditions;
+import org.apache.polaris.core.entity.PolarisEntity;
+import org.apache.polaris.core.policy.PolicyEntity;
+import org.apache.polaris.core.policy.PredefinedPolicyTypes;
+import org.apache.polaris.core.policy.validator.datacompaction.DataCompactionPolicyValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Validates a given {@link PolicyEntity} against its defined policy type.
+ *
+ * <p>This class maps the policy type code from the {@code PolicyEntity} to a predefined policy
+ * type, then delegates parsing/validation to a specific validator implementation.
+ */
+public class GenericPolicyValidator {
+  private static final Logger LOGGER = LoggerFactory.getLogger(GenericPolicyValidator.class);
+
+  /**
+   * Validates the given policy.
+   *
+   * @param policy the policy entity to validate
+   * @throws InvalidPolicyException if the policy type is unknown or unsupported, or if the policy
+   *     content is invalid
+   */
+  public static void validate(PolicyEntity policy) {
+    Preconditions.checkNotNull(policy, "Policy must not be null");
+
+    var type = PredefinedPolicyTypes.fromCode(policy.getPolicyTypeCode());
```

Comment:
```suggestion
    var type = policy.getPolicyType()
```
I think we can utilize the `getPolicyType()` here. The method will return a type or throw an error for corrupted PolicyEntity. So we can remove the Preconditions check below

Reply:
Got it, thanks for the explanation!

Reply:
Wondering if we should use the name of policy type in the `switch` clause, as customized type won't presented as an enum.

Reply:
I think for custom type we should rely on the `default` case to load their corresponding validator impl?

Reply:
we still need to distinguish the different customized types, right?